packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -1040,10 +1040,10 @@ export class Project extends ProjectBase {
         : new NoArtifacts());
     const artifactsConfig = artifacts.bind(this, this);
 
-    const cache = props.cache || Cache.none();
+    const cache = props.cache;
```

Comment:
The fact that no cache has been provided and the cache mode is left "unchanged" shouldn't even be a mode of operation, so I'm not convinced we should support it.

This PR is fixing the transition:

```ts
  cache: Cache.s3(...)
// ↓
  cache: Cache.none()
```

But the transition:

```ts
  cache: Cache.s3(...)
// ↓
  cache: undefined,
```

is equally valid and should *also* result in caching being turned off.

We should just change the rendering of `Cache.none()`, not change whether or not we render something.

Reply:
Sorry, I'm not sure I understand what the ask here is 😛.

Are you saying we should _always_ render `{ type: 'NO_CACHE' }` in the resulting CloudFormation template, even if the `cache` property of the L2 is `undefined`?

Reply:
Discussed offline. Yep, that's what the ask was 🙂. Will implement in the next revision.